### PR TITLE
added model User, service and dao layers for it and AuthenticationSer…

### PR DESCRIPTION
**Pull request by Nick97-git**
…vice

### CODE REVIEW

File: src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    public Optional<User> findByEmail(String email) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<User> criteriaQuery = criteriaBuilder
+                    .createQuery(User.class);
+            Root<User> root = criteriaQuery.from(User.class);
+            Predicate predicateEmail = criteriaBuilder.equal(root.get("email"), email);
+            criteriaQuery.where(predicateEmail);
+            User user = session.createQuery(criteriaQuery).uniqueResult();
+            return user != null ? Optional.of(user) : Optional.empty();
```
Comment (boroda4436) [inline src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java:54]:
```suggestion
            return Optional.ofNullable(user);
```

File: src/main/java/com/dev/cinema/service/impl/UserServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+@Service
+public class UserServiceImpl implements UserService {
+    @Inject
+    private UserDao userDao;
+
+    @Override
+    public User add(User user) {
+        user.setSalt(HashUtil.getSalt());
+        user.setPassword(HashUtil.hashPassword(user.getPassword(), user.getSalt()));
```
Comment (boroda4436) [inline src/main/java/com/dev/cinema/service/impl/UserServiceImpl.java:19]: Перенеси в реєстрацію

File: src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java (hunk 1 above)
Comment (Nick97-git) [reply src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java]: done

File: src/main/java/com/dev/cinema/service/impl/UserServiceImpl.java (hunk 2 above)
Comment (Nick97-git) [reply src/main/java/com/dev/cinema/service/impl/UserServiceImpl.java]: done

**Pull request merged**